### PR TITLE
20 Chain Compatibility

### PR DESCRIPTION
**Pull request by emilypi**

### CODE REVIEW

File: lib/ChainwebData/Genesis.hs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+( genesisHeight
+) where
+
+import Chainweb.Api.ChainId (ChainId(..))
+
+
+genesisHeight :: ChainId -> Int
+genesisHeight (ChainId c)
+  | c `elem` [0..9] = 0
+  | c `elem` [10..19] = 852_054
```
Comment (mightybyte) [inline lib/ChainwebData/Genesis.hs:12]: I don't think we should hard code these.  In the future we will probably start running chainweb-data for other networks like testnet, so we should get this information from the node.  It is available from the `/info` endpoint which is already being queried and made available in the environment.

**Review by mightybyte — CHANGES_REQUESTED** [review]

File: cabal.project [hunk 2]
```diff
@@ -0,0 +1,41 @@
+packages: chainweb-data.cabal
+
+package aeson
+    flags: +cffi
+
+source-repository-package
+    type: git
+    location: https://github.com/kadena-io/pact.git
+    tag: 94cfeb57e2f90c299fc27cc3e72f6b6aa9109c06
```
[the remark this answers is not in the document]
Comment (emilypi) [reply cabal.project:9]: I grabbed it from the `stack.yaml` dep. I'll bump to the kpkgs version.

[the remark this answers is not in the document]
Comment (emilypi) [reply cabal.project:9]: Looks like it was the nix dep that was out of sync with `kpkgs`! I bumped it in the latest commit, along with `gargoyle`.

[the remark this answers is not in the document]
Comment (emilypi) [reply cabal.project:9]: Fixed

File: lib/ChainwebData/Genesis.hs (hunk 1 above)
Comment (emilypi) [reply lib/ChainwebData/Genesis.hs]: This now uses the environment `NodeInfo` object to determine lower bounds for genesis.

**Review by mightybyte — APPROVED** [review]

**Pull request merged**